### DISCUSSION TITLE
ESP32-S3 OneWire

### DISCUSSION THREAD

**Discussion by markusschweitzer** [nanoframework/Home#1521] [Q&A] [answered]
Hi, I recently bought a ESP32-S3-DevKitC-1 N16R8 and try to get onewire working there.
At first I found out that the library is disabled per default for the ESP32-S3 target. Therefore I added it to CMakePresets.json and compiled the ESP32_S3 target via the dev container. 
But when I tried it, it still doesnt work (deployed the sample, but the `oneWire.FindAllDevices();` call only returns a 000000000 device. (It gets also returned when I disconnect the pins)

In `target_nf_dev_onewire_config.h` of the target I saw that UART2 is used. I tried severall other urats and pin combinations but  `oneWire.FindAllDevices();` always behaves the same.

Is there any way to get this working? I guess just a little hint is needed :)

best regards

**Comment by markusschweitzer** [comment, marked as answer]
Nevermind I figured it out (al lthe changes I made): 

* Add `API_nanoFramework.Device.OneWire": "ON"` to `CMAkePresets.json`
* In `target_nf_dev_onewire_config.h` set the UARt to `UART_NUM1` and `RX_PIN` to 18 and `TX_PIN` to 17
* In my project add ```Configuration.SetPinFunction(17, DeviceFunction.COM2_RX);
Configuration.SetPinFunction(18, DeviceFunction.COM2_TX);```

This made it work for me :) Leaving this here for other people who came looking for this

**Reply by Ellerbach** [reply]
Thanks for sharing!
